tools/deploy.fsx

```diff
@@ -0,0 +1,140 @@
+let cliDir = "./src"
+let testDir = "./test"
+let cliProj = cliDir</>"ts2fable.fsproj"
+let runDotnet workingDir args =
+    let result =
+        Process.ExecProcess (fun info ->
+        { info with 
+            FileName = dotnetExePath
+            WorkingDirectory = workingDir 
+            Arguments =  args }) TimeSpan.MaxValue
+    if result <> 0 then failwithf "dotnet %s failed" args 
+
+let yarn args = 
+    run yarnTool "./" args
+let node args = 
+    run nodeTool "./" args   
+
+let npm args = 
+    run npmTool "./" args   
+
+
+Target.Create "InstallDotNetCore" (fun _ ->
+    DotNetCliInstall Release_2_1_4
+    dotnetExePath <- DotNetInfoOptions.Create().Common.DotNetCliPath
+)
+
+Target.Create "YarnInstall" (fun _ ->
+    yarn "install"
+)
+
+Target.Create "Restore" (fun _ ->
+    DotNetRestore(id) (toolDir</>"DotnetCLI.fsproj")
+    DotNetRestore(id) (testDir</>"test.fsproj")
+    DotNetRestore(id) cliProj
+)
+
+Target.Create "BuildCli" (fun _ ->
+    runDotnet toolDir <| sprintf "fable yarn-fable-splitter -- %s  --config %s" cliProj (toolDir</>"splitter.config.js")
+)
+
+Target.Create "RunCli" (fun _ ->
+    node <| sprintf "%s cliTest" (distDir</>"ts2fable.js")
+)
+
+Target.Create "RestoreTestCompile" (fun _ ->
+    DotNetRestore(id) (testCompileTool</>"test-compile.fsproj")
+)
+
+Target.Create "BuildTest" (fun _ ->
+    runDotnet toolDir "fable webpack -- --config webpack.config.test.js"
+)
+
+Target.Create "RunTest" (fun _ ->
+    match buildServer with 
+    | AppVeyor -> node <| sprintf "%s --reporter mocha-appveyor-reporter %s" mochaPath (buildDir</>"test.js" |> Path.getFullName)
+    | _ -> node <| sprintf "%s %s" mochaPath (buildDir</>"test.js" |> Path.getFullName)
+)
+
+Target.Create "Publish" (fun _ ->
+    match buildServer with 
+    | AppVeyor -> 
+        node (toolDir</>"build-update.package.js")
+        let version = sprintf "0.6.0-build.%s" buildVersion
+        yarn <| sprintf "version --new-version %s --no-git-tag-version" version
+        npm "pack"
+        
+        let repoName = environVar "appveyor_repo_name"
+        let repoBranch = environVar "appveyor_repo_branch"
+        
+        if repoName = "fable-compiler/ts2fable" && repoBranch = "master" then
+            let line = sprintf "//registry.npmjs.org/:_authToken=%s\n" <| environVar "npmauthtoken`n"
+            let npmrc = (GetFolderPath UserProfile)</>".npmrc"
+            printfn "Auth token is %s" line
+            printfn "npmrc path is %s" npmrc
+            File.writeNew npmrc [line]
+            npm "whoami"
+            yarn <| sprintf "publish ts2fable-%s.tgz --new-version %s --tag next" version version
+    | _ ->  ()
+)
+
+Target.Create "Deploy" Target.DoNothing    
+
+"Deploy"
+    <== [ "InstallDotNetCore"
+          "YarnInstall"
+          "Restore"
+          "BuildTest"
+          "RunTest" 
+          "BuildCli"
+          "RunCli"
+          "RestoreTestCompile"
+          "Publish" ]
+
+Target.RunOrDefault "Deploy"
```

Comment:
With this the CI server will try to publish a new version to npm for every push (including PRs as it's happening now). @ctaggart how are you triggering deployments in Appveyor now?

Reply:
I collected all of appveyor server [Environments](https://ci.appveyor.com/project/fable-compiler/ts2fable/build/0.6.0-build.200)
And find the `APPVEYOR_PULL_REQUEST_HEAD_REPO_NAME` should be 
`fable-compiler/ts2fable`(current is `humhei/ts2fable`)

Reply:
Currently a successful build on master will use the `npmauthtoken` environment variable to publish. 
![image](https://user-images.githubusercontent.com/80104/36520511-61a8205c-175f-11e8-897c-8dd9425ee283.png)

Reply:
Nevermind, looks like @humhei figured it out.